File: java/siebog/agents/MobileAgent.java
package siebog.agents;

import siebog.agentmanager.XjafAgent;
import siebog.agentmanager.Agent;

import siebog.messagemanager.ACLMessage;
import javax.ejb.Remote;
import javax.ejb.Stateless;
import siebog.messagemanager.Performative;
import util.LoggerUtil;

@Stateless
@Remote(Agent.class)
public class MobileAgent extends XjafAgent {
	private static final long serialVersionUID = 1L;

	@Override
	protected void onInit(AgentInitArgs args){
		
	}

	@Override
	protected void onMessage(ACLMessage msg) {
		switch (msg.performative) {
			case REQUEST:			
				LoggerUtil.log("Received message to move to host: " +msg.content);
				this.move(msg.content);
				break;
			case RESUME:			
				LoggerUtil.log("Arrived at host: " +msg.content, true);
				break;
			default:			
				LoggerUtil.log("Wrong performative. Send me REQUEST to move to other node.", true);
		}
	}

	@Override
	protected void doTasks(AgentInitArgs args) {
		
	}
	
	/*
	public static void main(String[] args) {
		SiebogClient.connect("localhost");
		AgentBuilder.module("dnars-web").ejb(MobileAgent.class).randomName().start();
	} */
}

File: js/siebog/agents/MobileAgent.js
importScripts("/siebog-war/js/radigost/agent.js");

function MobileAgent() {
	
};

MobileAgent.prototype.onInit = function(args, _post, _log, _moveToServer) {
	
};

MobileAgent.prototype.onArrived = function(host, isServer) {		
	console.log("Arrived at host: " +msg.content, true);
};

MobileAgent.prototype.onMessage = function(msg) {
	switch (msg.performative) {
		case "REQUEST":			
			console.log("Received message to move to host: " +msg.content);
			this.moveToServer(msg.content);
			break;
		default:			
			console.log("Wrong performative. Send me REQUEST to move to other node.", true);
	}
};


MobileAgent.prototype = new Agent();

/*
 * This links this agent to its Worker. 
 */
setAgentInstance(new MobileAgent());